pkg/controller/scheduler.go

```diff
@@ -322,32 +325,29 @@ func (c *Controller) advanceCanary(name string, namespace string) {
 
 	// route traffic back to primary if analysis has succeeded
 	if cd.Status.Phase == flaggerv1.CanaryPhasePromoting {
+		//start traffic shift only when the primary is ready
+		if err := canaryController.IsPrimaryReady(cd); err != nil {
+			c.recordEventWarningf(cd, "%v", err)
+			return
+		}
```

Comment:
We already check for that in the beginning: https://github.com/fluxcd/flagger/blob/560f884cc0046cb12c39d73b05982bbd0401d4aa/pkg/controller/scheduler.go#L250-L256